refactor: switch language handling to i18next

Quantumlyy opened the pull request.

Quantumlyy committed:
chore: add i18next deps

Quantumlyy committed:
chore: switch klasa to i18next branch

Quantumlyy committed:
refactor: remove language tests

Quantumlyy committed:
feat: add the base language handler

Quantumlyy committed:
refactor: commit

kyranet committed:
chore: first step at moving language to json files

favna committed:
chore: bump packages

closes #1423

favna committed:
refactor: use fs/promises everywheree

depfu[bot] committed:
Upgrade discord-api-types: 0.9.1 → 0.10.0 (major) (#1425)

Co-authored-by: depfu[bot] <23717796+depfu[bot]@users.noreply.github.com>

depfu[bot] committed:
Update all Yarn dependencies (2020-12-13) (#1428)

Co-authored-by: depfu[bot] <23717796+depfu[bot]@users.noreply.github.com>

kyranet committed:
chore: port more keys

kyranet committed:
chore: translated more keys

favna committed:
fix: formatting fixes + globals and system keys

kyranet committed:
chore: move almost all of the remaining keys

vladfrangu commented on src/languages/en-US/commands/admin.json:
```diff
@@ -0,0 +1,71 @@
+{
+	"commandConfNoKey": "You must provide a key",
+	"commandConfNoValue": "You must provide a value",
+	"commandConfGuarded": "${toTitleCase(name)} may not be disabled.",
```
toTitleCase

vladfrangu commented on src/languages/en-US/commands/admin.json:
```diff
@@ -0,0 +1,71 @@
+	"commandRolesetResetGroup": "{{GREENTICK}} Successfully removed the roleset `{{name}}` from this server.",
+	"commandRolesetUpdated": "The {{name}} unique role set has been updated.",
+	"commandRolesetNoRolesets": "You have no rolesets.",
+	"commandConfMenuNopermissions": "I need the {{this.PERMISSIONS.ADD_REACTIONS}} and {{this.PERMISSIONS.MANAGE_MESSAGES}} permissions to be able to run the menu.",
```
Perms

Quantumlyy committed:
feat: update message extension to new format

vladfrangu committed:
chore: work it (#1430)

favna committed:
chore: obligatory placeholder commit message

Quantumlyy committed:
refactor: yea i forgot em

favna committed:
Finish up language keys

Co-authored-by: Antonio Román <kyradiscord@gmail.com>

favna committed:
top language keys namespaced

Co-authored-by: Antonio Román <kyradiscord@gmail.com>

favna committed:
namespaces go brrrr

favna committed:
regen lockfile

favna committed:
fix: remaining function calls to use i18n formatters

favna committed:
chore: starting the service is always better

dependabot[bot] committed:
build(deps): bump node-notifier from 8.0.0 to 8.0.1 (#1432)

Bumps [node-notifier](https://github.com/mikaelbr/node-notifier) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/mikaelbr/node-notifier/releases)
- [Changelog](https://github.com/mikaelbr/node-notifier/blob/v8.0.1/CHANGELOG.md)
- [Commits](https://github.com/mikaelbr/node-notifier/compare/v8.0.0...v8.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

favna committed:
chore: prettier up the repo

favna committed:
chore: bump lockfile

favna committed:
Update Crowdin configuration file

favna committed:
Update crowdin.yml

favna committed:
New Crowdin updates (#1434)

favna committed:
chore: update crowdin yaml

favna committed:
New Crowdin updates (#1435)

[skip ci]

favna committed:
chore: update crowdin.yml

depfu[bot] committed:
Upgrade @types/diff: 4.0.2 → 5.0.0 (major) (#1436)

Co-authored-by: depfu[bot] <23717796+depfu[bot]@users.noreply.github.com>

depfu[bot] committed:
chore(package): update all Yarn dependencies (2020-12-27) (#1437)

favna committed:
fix(i18n): update deejay -> DJ in en-US

favna committed:
New Crowdin updates (#1439)

NM-EEA-Y committed:
New Crowdin updates (#1440)

depfu[bot] committed:
Update discord-api-types to version 0.12.0 (#1441)

depfu[bot] committed:
Update all Yarn dependencies (2021-01-03) (#1442)

Co-authored-by: depfu[bot] <23717796+depfu[bot]@users.noreply.github.com>

NM-EEA-Y committed:
feat(languages): add ru-RU (#1443)

NM-EEA-Y committed:
New Crowdin updates (#1444)

NM-EEA-Y committed:
New Crowdin updates (#1446)

NM-EEA-Y committed:
New Crowdin updates (#1447)

Quantumlyy committed:
refactor: use plugin-i18next over in17n (#1445)

NM-EEA-Y committed:
New Crowdin updates (#1448)

kyranet committed:
chore: many fixes

kyranet committed:
chore: fix building commit

NM-EEA-Y committed:
New Crowdin updates (#1449)

favna committed:
chore: be sanic before depfu on dependencies

favna committed:
test: fix tests

NM-EEA-Y committed:
New Crowdin updates (#1450)

kyranet committed:
chore: updated klasa

favna committed:
chore: remove unused key

favna committed:
fix(i18n): improve some base strings

favna committed:
chore: don’t mess with voodoo

vladfrangu commented on crowdin.yml:
```diff
@@ -0,0 +1,8 @@
+project_id: '434340'
+api_token_env: 'CROWDIN_PERSONAL_TOKEN'
+commit_message: 'fix(i18n-crowdin): new translations for %original_file_name% (%language%) [skip ci]'
```
```suggestion
commit_message: 'chore(i18n-crowdin): new translations for %original_file_name% (%language%) [skip ci]'
```

vladfrangu commented on src/commands/Admin/roleset.ts:
```diff
@@ -63,11 +63,11 @@ export default class extends SkyraCommand {
+				roles: t(
 					roleSets.map((role) => role.name),
-					language.get(LanguageKeys.Globals.And)
+					t(LanguageKeys.Globals.And)
```
Isn't this.. wrong?

Stitch07 replied:
Yeah, it should use the andList formatter.

vladfrangu commented on src/commands/Admin/roleset.ts:
```diff
@@ -87,37 +87,37 @@ export default class extends SkyraCommand {
+				roles: t(
 					roles.map((role) => role.name),
-					language.get(LanguageKeys.Globals.And)
+					t(LanguageKeys.Globals.And)
```
Ditto

vladfrangu commented on src/commands/Admin/roleset.ts:
```diff
@@ -144,14 +144,14 @@ export default class extends SkyraCommand {
 		// Get all rolesets from settings
 		const allRolesets = await message.guild.readSettings(GuildSettings.Roles.UniqueRoleSets);
-		if (!allRolesets.length) return message.sendLocale(LanguageKeys.Commands.Admin.RoleSetNoRolesets);
+		if (!allRolesets.length) return message.sendTranslated(LanguageKeys.Commands.Admin.RoleSetNoRolesets);
```
nit: Usually this uses throw but ok

vladfrangu commented on src/commands/Announcement/announcement.ts:
```diff
@@ -27,32 +27,32 @@ export default class extends SkyraCommand {
 
 		const role = await announcementCheck(message);
-		const header = language.get(LanguageKeys.Commands.Announcement.Announcement, { role: role.toString() });
+		const header = t(LanguageKeys.Commands.Announcement.Announcement, { role: role.toString() });
```
Creative naming for the header...please rename the key to Header or AnnouncementHeader or something more...descriptive..

vladfrangu commented on src/commands/Fun/changemymind.ts:
```diff
@@ -13,8 +13,8 @@ import { join } from 'path';
 	cooldown: 10,
-	description: (language) => language.get(LanguageKeys.Commands.Fun.ChangemymindDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Fun.ChangemymindExtended),
+	description: LanguageKeys.Commands.Fun.ChangemymindDescription,
```
Capitalization of ChangeMyMind

vladfrangu commented on src/commands/Fun/escaperope.ts:
```diff
@@ -10,19 +10,19 @@ import { KlasaMessage } from 'klasa';
 				.setColor(await DbSet.fetchColor(message))
 				.setImage(CdnUrls.EscapeRopeGif)
-				.setDescription(language.get(LanguageKeys.Commands.Fun.EscaperopeOutput, { user: message.author.toString() }))
+				.setDescription(t(LanguageKeys.Commands.Fun.EscaperopeOutput, { user: message.author.toString() }))
```
Capitalization of EscapeRope

vladfrangu commented on src/commands/Fun/escaperope.ts:
```diff
@@ -10,19 +10,19 @@ import { KlasaMessage } from 'klasa';
 	cooldown: 60,
-	description: (language) => language.get(LanguageKeys.Commands.Fun.EscaperopeDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Fun.EscaperopeExtended)
+	description: LanguageKeys.Commands.Fun.EscaperopeDescription,
```
Ditto

vladfrangu commented on src/commands/Fun/markov.ts:
```diff
@@ -22,14 +23,14 @@ export default class extends SkyraCommand {
-	private readonly kProcess: (message: GuildMessage, language: Language, markov: Markov) => Promise<MessageEmbed>;
+	private readonly kProcess: (message: GuildMessage, language: TFunction, markov: Markov) => Promise<MessageEmbed>;
 
 	public constructor(store: CommandStore, file: string[], directory: string) {
```
Move to decorator

vladfrangu commented on src/commands/Fun/peepolove.ts:
```diff
@@ -12,8 +12,8 @@ import { join } from 'path';
 	cooldown: 10,
-	description: (language) => language.get(LanguageKeys.Commands.Fun.PeepoloveDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Fun.PeepoloveExtended),
+	description: LanguageKeys.Commands.Fun.PeepoloveDescription,
```
PeepoLove

vladfrangu reviewed (changes requested):
For now, mobile phone reviews let's go

Soumil-07 committed:
chore: requested changes

Soumil-07 committed:
refactor: rename announcement key to announcementHeader

Part of Vlad's original requested change

NM-EEA-Y committed:
New Crowdin updates (#1451)

kyranet committed:
fix: resolved multiple critical bugs

kyranet committed:
fix(giveaway): correct i18n keys

kyranet committed:
fix(markov): resolve lint error

kyranet committed:
fix: more fixes

kyranet committed:
chore: more fixes

kyranet committed:
fix: even more fixes

kyranet committed:
feat: shamelessly plug the translation dashboard

kyranet committed:
fix: fixes and fixes

kyranet committed:
fix: fix of fixes

kyranet committed:
fix: fixes within fixes within fixes

This is a reference about my mom wearing a bag within a bag within a bag within a bag, when I went shopping yesterday

kyranet committed:
fix: another

kyranet committed:
fix: more and more fixes

kyranet committed:
fix: fixing fixes

NM-EEA-Y committed:
New Crowdin updates (#1452)

kyranet committed:
fix: fix fix fix

kyranet committed:
fix: more fixes

NM-EEA-Y committed:
New Crowdin updates (#1453)

favna committed:
chore: formatting the repo

favna committed:
chore: add index file for setup directory

favna committed:
fix(i18n): update some base strings

favna committed:
chore: ignore language files from prettier

kyranet committed:
refactor: simplify language files

kyranet committed:
fix: add missing line

kyranet committed:
fix: make more keys plain

kyranet committed:
fix: resolved test bugs

NM-EEA-Y committed:
New Crowdin updates (#1454)

kyranet committed:
refactor: use Intl

favna committed:
ci: setup resolving commit message for pull_request

favna committed:
ci(codeql): also setup skipping codeql

favna committed:
ci: cleanup pre_ci

kyranet committed:
fix: fixes within fixes

favna committed:
ci: improve pre_ci job

kyranet committed:
refactor: code quality +100

favna committed:
chore: improve some en-US strings

kyranet committed:
fix: ignore rule globally

kyranet committed:
fix: fallback to en-US if assets are missing

NM-EEA-Y committed:
New Crowdin updates (#1455)

favna committed:
chore: forgot we're not using a smart language

vladfrangu commented on src/commands/Admin/roleset.ts:
```diff
@@ -144,14 +138,14 @@ export default class extends SkyraCommand {
 		// Get all rolesets from settings
 		const allRolesets = await message.guild.readSettings(GuildSettings.Roles.UniqueRoleSets);
-		if (!allRolesets.length) return message.sendLocale(LanguageKeys.Commands.Admin.RoleSetNoRolesets);
+		if (!allRolesets.length) throw message.resolveKey(LanguageKeys.Commands.Admin.RoleSetNoRolesets);
```
Caught ya!
```suggestion
		if (!allRolesets.length) throw message.resolveKey(LanguageKeys.Commands.Admin.RoleSetNoRoleSets);
```

vladfrangu commented on src/commands/Anime/anime.ts:
```diff
@@ -94,27 +91,27 @@ export default class extends RichDisplayCommand {
 					)
 					.setThumbnail(entry.posterImage?.original ?? '')
-					.addField(embedData.type, language.get(LanguageKeys.Commands.Anime.AnimeTypes)[type.toUpperCase()] || type, true)
+					.addField(embedData.type, t(LanguageKeys.Commands.Anime.AnimeTypes)[type.toUpperCase()] || type, true)
```
Does this need to returnObjects?

vladfrangu commented on src/commands/Anime/manga.ts:
```diff
@@ -94,25 +91,25 @@ export default class extends RichDisplayCommand {
 					)
 					.setThumbnail(entry.posterImage?.original || '')
-					.addField(embedData.type, language.get(LanguageKeys.Commands.Anime.MangaTypes)[type.toUpperCase()] || type, true)
+					.addField(embedData.type, t(LanguageKeys.Commands.Anime.MangaTypes)[type.toUpperCase()] || type, true)
```
Ditto

vladfrangu commented on src/commands/Developers/yarn.ts:
```diff
@@ -5,73 +5,73 @@ import { YarnPkg } from '#lib/types/definitions/Yarnpkg';
+import { KlasaMessage } from 'klasa';
 
 @ApplyOptions<SkyraCommandOptions>({
 	aliases: ['npm', 'npm-package', 'yarn-package'],
```
Why not

```suggestion
	aliases: ['npm', 'npm-package', 'yarn-package', 'pnpm', 'pnpm-package'],
```

vladfrangu commented on src/commands/Fun/xkcd.ts:
```diff
@@ -4,50 +4,48 @@ import { LanguageKeys } from '#lib/types/namespaces/LanguageKeys';
+		const comicNumber = await this.getNumber(query, t);
 		const comic = await fetch<XkcdResultOk>(`https://xkcd.com/${comicNumber}/info.0.json`, FetchResultTypes.JSON).catch(() => {
-			throw language.get(LanguageKeys.Commands.Fun.XkcdNotfound);
+			throw t(LanguageKeys.Commands.Fun.XkcdNotfound);
```
nit: I assume this is a sanity check..?

kyranet replied:
Yeah, to send an user friendly message when XKCD goes down, fixed the typo tough.

vladfrangu commented on src/commands/GameIntegration/brawlstars.ts:
```diff
@@ -49,8 +50,8 @@ export interface BrawlStarsGIData {
-	description: (language) => language.get(LanguageKeys.Commands.GameIntegration.BrawlstarsDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.GameIntegration.BrawlstarsExtended),
+	description: LanguageKeys.Commands.GameIntegration.BrawlstarsDescription,
+	extendedHelp: LanguageKeys.Commands.GameIntegration.BrawlstarsExtended,
```
BrawlStarts
Capitalization plz

vladfrangu commented on src/commands/GameIntegration/brawlstars.ts:
```diff
@@ -61,59 +62,58 @@ export interface BrawlStarsGIData {
-		const languageString = language.name;
+	private async buildPlayerEmbed(message: KlasaMessage, t: TFunction, player: BrawlStars.Player) {
+		const titles = t(LanguageKeys.Commands.GameIntegration.BrawlstarsPlayerEmbedTitles);
+		const fields = t(LanguageKeys.Commands.GameIntegration.BrawlstarsPlayerEmbedFields);
```
Capitalization plz

vladfrangu commented on src/commands/GameIntegration/clashofclans.ts:
```diff
@@ -23,8 +23,8 @@ const kFilterSpecialCharacters = /[^A-Z0-9]+/gi;
-	description: (language) => language.get(LanguageKeys.Commands.GameIntegration.ClashofclansDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.GameIntegration.ClashofclansExtended),
+	description: LanguageKeys.Commands.GameIntegration.ClashofclansDescription,
+	extendedHelp: LanguageKeys.Commands.GameIntegration.ClashofclansExtended,
```
Ditto

vladfrangu commented on src/commands/GameIntegration/clashofclans.ts:
```diff
@@ -132,11 +132,11 @@ export default class extends RichDisplayCommand {
 
 		for (const clan of clans) {
-			const titles = language.get(LanguageKeys.Commands.GameIntegration.ClashofclansClanEmbedTitles);
+			const titles = t(LanguageKeys.Commands.GameIntegration.ClashofclansClanEmbedTitles);
```
.

vladfrangu commented on src/commands/Games/coinflip.ts:
```diff
@@ -25,10 +26,10 @@ const enum CoinType {
-			const face = language.get(LanguageKeys.Commands.Games.CoinFlipCoinNames).findIndex((coin) => coin.toLowerCase() === lArg);
-			if (face === -1) throw language.get(LanguageKeys.Commands.Games.CoinFlipInvalidCoinname, { arg });
+			const face = t(LanguageKeys.Commands.Games.CoinFlipCoinNames).findIndex((coin) => coin.toLowerCase() === lArg);
+			if (face === -1) throw t(LanguageKeys.Commands.Games.CoinFlipInvalidCoinname, { arg });
```
Capitalization

vladfrangu commented on src/commands/Games/coinflip.ts:
```diff
@@ -64,45 +65,45 @@ export default class extends SkyraCommand {
-					language.get(LanguageKeys.Commands.Games.CoinFlipNoGuessDescription, {
-						result: language.get(LanguageKeys.Commands.Games.CoinFlipCoinNames)[result]
+					t(LanguageKeys.Commands.Games.CoinFlipNoGuessDescription, {
+						result: t(LanguageKeys.Commands.Games.CoinFlipCoinNames)[result]
```
returnObjects?

vladfrangu commented on src/commands/Games/coinflip.ts:
```diff
@@ -64,45 +65,45 @@ export default class extends SkyraCommand {
-					language.get(won ? LanguageKeys.Commands.Games.CoinFlipWinDescription : LanguageKeys.Commands.Games.CoinFlipLoseDescription, {
-						result: language.get(LanguageKeys.Commands.Games.CoinFlipCoinNames)[result]
+					t(won ? LanguageKeys.Commands.Games.CoinFlipWinDescription : LanguageKeys.Commands.Games.CoinFlipLoseDescription, {
+						result: t(LanguageKeys.Commands.Games.CoinFlipCoinNames)[result]
```
returnObjects?

vladfrangu commented on src/commands/Games/coinflip.ts:
```diff
@@ -64,45 +65,45 @@ export default class extends SkyraCommand {
 							: LanguageKeys.Commands.Games.CoinFlipLoseDescriptionWithWager,
 						{
-							result: language.get(LanguageKeys.Commands.Games.CoinFlipCoinNames)[result],
+							result: t(LanguageKeys.Commands.Games.CoinFlipCoinNames)[result],
```
returnObjects?

vladfrangu commented on src/commands/Games/higherLower.ts:
```diff
@@ -174,7 +174,7 @@ export default class extends SkyraCommand {
 				break;
 			case HigherLowerReactions.Ok:
-				await game.response.edit(game.language.get(LanguageKeys.Commands.Games.HigherLowerNewround), { embed: null });
+				await game.response.edit(game.t(LanguageKeys.Commands.Games.HigherLowerNewround), { embed: null });
```
Capitalization

vladfrangu commented on src/commands/Games/slotmachine.ts:
```diff
@@ -10,8 +10,8 @@ import { KlasaMessage } from 'klasa';
-	description: (language) => language.get(LanguageKeys.Commands.Games.SlotmachineDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Games.SlotmachineExtended),
+	description: LanguageKeys.Commands.Games.SlotmachineDescription,
+	extendedHelp: LanguageKeys.Commands.Games.SlotmachineExtended,
```
Capitalization

vladfrangu commented on src/commands/Games/slotmachine.ts:
```diff
@@ -21,11 +21,11 @@ export default class extends SkyraCommand {
 
 		const [attachment, amount] = await new Slotmachine(message, wager, settings).run();
-		const titles = await message.fetchLocale(LanguageKeys.Commands.Games.SlotmachineTitles);
+		const titles = await message.resolveKey(LanguageKeys.Commands.Games.SlotmachineTitles);
```
Capitalization

vladfrangu commented on src/commands/General/Chat Bot Info/help.ts:
```diff
@@ -103,14 +103,14 @@ export default class extends SkyraCommand {
+			return message.channel.type === 'dm' ? response : await message.sendTranslated(LanguageKeys.Commands.General.HelpDm);
 		} catch {
-			return message.channel.type === 'dm' ? null : message.sendLocale(LanguageKeys.Commands.General.HelpNodm);
+			return message.channel.type === 'dm' ? null : message.sendTranslated(LanguageKeys.Commands.General.HelpNodm);
```
Capitalization

vladfrangu commented on src/commands/Management/Configuration/setPrefix.ts:
```diff
@@ -8,30 +8,30 @@ import { ApplyOptions } from '@skyra/decorators';
-	description: (language) => language.get(LanguageKeys.Commands.Management.SetprefixDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Management.SetprefixExtended),
+	description: LanguageKeys.Commands.Management.SetprefixDescription,
+	extendedHelp: LanguageKeys.Commands.Management.SetprefixExtended,
```
Capitalization

vladfrangu commented on src/commands/Misc/deletthis.ts:
```diff
@@ -14,8 +14,8 @@ import { join } from 'path';
-	description: (language) => language.get(LanguageKeys.Commands.Misc.DeletthisDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Misc.DeletthisExtended),
+	description: LanguageKeys.Commands.Misc.DeletthisDescription,
+	extendedHelp: LanguageKeys.Commands.Misc.DeletthisExtended,
```
Capitalizaton, DeletThis

vladfrangu commented on src/commands/Misc/goodnight.ts:
```diff
@@ -13,8 +13,8 @@ import { join } from 'path';
-	description: (language) => language.get(LanguageKeys.Commands.Misc.GoodnightDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Misc.GoodnightExtended),
+	description: LanguageKeys.Commands.Misc.GoodnightDescription,
+	extendedHelp: LanguageKeys.Commands.Misc.GoodnightExtended,
```
capitalization

vladfrangu commented on src/commands/Misc/goofytime.ts:
```diff
@@ -13,8 +13,8 @@ import { join } from 'path';
-	description: (language) => language.get(LanguageKeys.Commands.Misc.GoofytimeDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Misc.GoofytimeExtended),
+	description: LanguageKeys.Commands.Misc.GoofytimeDescription,
+	extendedHelp: LanguageKeys.Commands.Misc.GoofytimeExtended,
```
ditto

vladfrangu commented on src/commands/Misc/ineedhealing.ts:
```diff
@@ -13,8 +13,8 @@ import { join } from 'path';
-	description: (language) => language.get(LanguageKeys.Commands.Misc.IneedhealingDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Misc.IneedhealingExtended),
+	description: LanguageKeys.Commands.Misc.IneedhealingDescription,
+	extendedHelp: LanguageKeys.Commands.Misc.IneedhealingExtended,
```
ditto

vladfrangu commented on src/commands/Misc/randreddit.ts:
```diff
@@ -5,7 +5,8 @@ import { Reddit } from '#lib/types/definitions/Reddit';
+import { CommandStore } from 'klasa';
 
 export default class extends SkyraCommand {
 	private readonly kBlacklist = /nsfl|morbidreality|watchpeopledie|fiftyfifty|stikk/i;
```
Honestly, for future-proofing, make this an array so we can add more and more if needed (like r/the-donal-oh wait :3)

kyranet replied:
It's unnecessary, array or not, this is easy-to-read syntax.

vladfrangu commented on src/commands/Misc/thesearch.ts:
```diff
@@ -13,8 +13,8 @@ export default class extends SkyraCommand {
-			description: (language) => language.get(LanguageKeys.Commands.Misc.ThesearchDescription),
-			extendedHelp: (language) => language.get(LanguageKeys.Commands.Misc.ThesearchExtended),
+			description: LanguageKeys.Commands.Misc.ThesearchDescription,
+			extendedHelp: LanguageKeys.Commands.Misc.ThesearchExtended,
```
Capitalization

vladfrangu commented on src/commands/Moderation/Utilities/case.ts:
```diff
@@ -1,25 +1,22 @@
+	permissionLevel: PermissionLevels.Moderator,
+	requiredPermissions: ['EMBED_LINKS'],
+	runIn: ['text'],
+	usage: '<latest|case:integer{0,2147483647}>'
```
n/a: bold of you to assume that's a case someone won't get :)

kyranet replied:
FlareTV is the guild with the highest moderation case list, with a total of 15187 cases, after 2 years.

I doubt we'll have more than that by 2077, either way, by then, we'd most likely have switched this field to a BIGINT.

vladfrangu commented on src/commands/Moderation/removerole.ts:
```diff
@@ -9,8 +9,8 @@ import { Role, User } from 'discord.js';
-	description: (language) => language.get(LanguageKeys.Commands.Moderation.RemoveroleDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Moderation.RemoveroleExtended),
+	description: LanguageKeys.Commands.Moderation.RemoveroleDescription,
+	extendedHelp: LanguageKeys.Commands.Moderation.RemoveroleExtended,
```
Capitalization

vladfrangu commented on src/commands/Music/join.ts:
```diff
@@ -18,7 +19,7 @@ export default class extends MusicCommand {
 
 		// If the member is not in a voice channel then throw
-		if (!channel) throw await message.fetchLocale(LanguageKeys.Commands.Music.JoinNoVoicechannel);
+		if (!channel) throw await message.resolveKey(LanguageKeys.Commands.Music.JoinNoVoicechannel);
```
```suggestion
		if (!channel) throw await message.resolveKey(LanguageKeys.Commands.Music.JoinNoVoiceChannel);
```

vladfrangu commented on src/commands/Music/join.ts:
```diff
@@ -35,27 +36,27 @@ export default class extends MusicCommand {
 		}
 
-		return message.sendLocale(LanguageKeys.Commands.Music.JoinSuccess, [{ channel: `<#${channel.id}>` }]);
+		return message.sendTranslated(LanguageKeys.Commands.Music.JoinSuccess, [{ channel: `<#${channel.id}>` }]);
```
Odd choice to make a channel mention to a voice channel...

kyranet replied:
In case Discord implements a way to mention channels, it'll be better than plain text, and we won't have to change a thing.

Also, unlike sending the name, the channel's name updates, even if it doesn't mention.

vladfrangu commented on src/commands/Music/leave.ts:
```diff
@@ -24,6 +24,6 @@ export default class extends MusicCommand {
 		}
 
-		return message.sendLocale(LanguageKeys.Commands.Music.LeaveSuccess, [{ channel: `<#${channelID}>` }]);
+		return message.sendTranslated(LanguageKeys.Commands.Music.LeaveSuccess, [{ channel: `<#${channelID}>` }]);
```
Ditto

vladfrangu commented on src/commands/Music/promote.ts:
```diff
@@ -25,26 +25,21 @@ export default class extends MusicCommand {
-		const language = await message.fetchLanguage();
-		if (index < 0) throw language.get(LanguageKeys.Commands.Music.RemoveIndexInvalid);
+		const t = await message.fetchT();
+		if (index < 0) throw t(LanguageKeys.Commands.Music.RemoveIndexInvalid);
```
This isn't remove tho

vladfrangu commented on src/commands/Music/promote.ts:
```diff
@@ -25,26 +25,21 @@ export default class extends MusicCommand {
-						count: length
-					}
-				)
+			throw t(LanguageKeys.Commands.Music.RemoveIndexOutOfBounds, {
```
Also not remove

vladfrangu commented on src/commands/Pokemon/item.ts:
```diff
@@ -6,23 +6,24 @@ import { fetchGraphQLPokemon, getItemDetailsByFuzzy, parseBulbapediaURL } from '
-		const embedTranslations = language.get(LanguageKeys.Commands.Pokemon.ItemEmbedData, {
-			availableInGen8: language.get(itemDetails.isNonstandard === 'Past' ? LanguageKeys.Globals.No : LanguageKeys.Globals.Yes)
+		const embedTranslations = t(LanguageKeys.Commands.Pokemon.ItemEmbedData, {
+			availableInGen8: t(itemDetails.isNonstandard === 'Past' ? LanguageKeys.Globals.No : LanguageKeys.Globals.Yes)
```
n/a since not an issue in skyra but @Favna pls. isNonStandard

vladfrangu commented on src/commands/Pokemon/type.ts:
```diff
@@ -58,24 +58,24 @@ const kPokemonTypes = new Set([
-			throw language.get(LanguageKeys.Commands.Pokemon.TypeQueryFail, {
-				types: types.map((val) => `\`${val}\``).join(` ${language.get(LanguageKeys.Globals.And)} `)
+			throw t(LanguageKeys.Commands.Pokemon.TypeQueryFail, {
+				types: types.map((val) => `\`${val}\``).join(` ${t(LanguageKeys.Globals.And)} `)
```
Shouldn't this.. use.. listing? ala `item1, item2, ..., itemn-1 and itemn`?

vladfrangu commented on src/commands/Social/Member Management/social.ts:
```diff
@@ -46,23 +44,19 @@ export default class extends SkyraCommand {
 		const { members } = await DbSet.connect();
 		const settings = await members.findOne({ where: { userID: user.id, guildID: message.guild.id } });
-		if (!settings) throw await message.fetchLocale(LanguageKeys.Commands.Social.SocialMemberNotexists);
+		if (!settings) throw await message.resolveKey(LanguageKeys.Commands.Social.SocialMemberNotexists);
```
```suggestion
		if (!settings) throw await message.resolveKey(LanguageKeys.Commands.Social.SocialMemberNotExists);
```

vladfrangu commented on src/commands/Social/Profile Management/banner.ts:
```diff
@@ -70,58 +70,58 @@ export default class extends SkyraCommand {
 			const banner = this.banners.get(arg);
 			if (banner) return banner;
-			throw await message.fetchLocale(LanguageKeys.Commands.Social.BannerNotexists, {
+			throw await message.resolveKey(LanguageKeys.Commands.Social.BannerNotexists, {
```
```suggestion
			throw await message.resolveKey(LanguageKeys.Commands.Social.BannerNotExists, {
```

vladfrangu commented on src/commands/Social/Profile Management/banner.ts:
```diff
@@ -175,13 +175,16 @@ export default class extends SkyraCommand {
-				new MessageEmbed({ description: pickRandom(language.get(LanguageKeys.System.Loading)), color: BrandingColors.Secondary })
+				new MessageEmbed({
+					description: pickRandom(t(LanguageKeys.System.Loading)),
+					color: BrandingColors.Secondary
```
n/a but this inconsistency is annoying..

vladfrangu commented on src/commands/Social/autorole.ts:
```diff
@@ -39,16 +39,16 @@ const SORT = (x: RolesAuto, y: RolesAuto) => Number(x.points > y.points) || Numb
-		const output = await message.guild.writeSettings((settings) => {
-			const language = settings.getLanguage();
+		const output = await message.guild.writeSettings(async (settings) => {
+			const t = await message.fetchT();
```
don't you have settings.getLanguage()?

kyranet replied:
Somebody did a fucky whoopsy during LS, thanks for catching this deadlock

vladfrangu commented on src/commands/Social/level.ts:
```diff
@@ -47,7 +47,8 @@ export default class extends SkyraCommand {
 		]);
 
-		const title = await message.fetchLocale(LanguageKeys.Commands.Social.Level);
+		const t = await message.fetchT();
+		const title = t(LanguageKeys.Commands.Social.Level);
```
isn't there resolveKey?

vladfrangu commented on src/commands/Social/marry.ts:
```diff
@@ -80,36 +80,31 @@ export default class extends RichDisplayCommand {
 			switch (answer) {
 				case YesNoAnswer.Timeout:
-					return message.sendLocale(LanguageKeys.Commands.Social.MarryNoreply);
+					return message.send(t(LanguageKeys.Commands.Social.MarryNoreply));
```
MarryNoReply

vladfrangu commented on src/commands/Social/mylevel.ts:
```diff
@@ -10,8 +10,8 @@ export default class extends SkyraCommand {
-			description: (language) => language.get(LanguageKeys.Commands.Social.MylevelDescription),
-			extendedHelp: (language) => language.get(LanguageKeys.Commands.Social.MylevelExtended),
+			description: LanguageKeys.Commands.Social.MylevelDescription,
+			extendedHelp: LanguageKeys.Commands.Social.MylevelExtended,
```
MyLevel

vladfrangu commented on src/commands/Social/mylevel.ts:
```diff
@@ -23,18 +23,18 @@ export default class extends SkyraCommand {
 		const nextRole = this.getLatestRole(memberPoints, roles);
 		const title = nextRole
-			? `\n${language.get(LanguageKeys.Commands.Social.MylevelNext, {
+			? `\n${t(LanguageKeys.Commands.Social.MylevelNext, {
```
MyLevel

vladfrangu commented on src/commands/Social/profile.ts:
```diff
@@ -52,7 +52,8 @@ export default class extends SkyraCommand {
 		]);
 
-		const title = await message.fetchLocale(LanguageKeys.Commands.Social.Profile);
+		const t = await message.fetchT();
+		const title = t(LanguageKeys.Commands.Social.Profile);
```
ditto, isn't there resolveKey?

vladfrangu commented on src/commands/Social/remindme.ts:
```diff
@@ -30,8 +29,8 @@ interface ReminderScheduledTask extends ScheduleEntity {
-	description: (language) => language.get(LanguageKeys.Commands.Social.RemindmeDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Social.RemindmeExtended),
+	description: LanguageKeys.Commands.Social.RemindmeDescription,
+	extendedHelp: LanguageKeys.Commands.Social.RemindmeExtended,
```
RemindMe

vladfrangu commented on src/commands/Social/remindme.ts:
```diff
@@ -71,23 +70,23 @@ interface ReminderScheduledTask extends ScheduleEntity {
 				case Actions.Show:
 				case Actions.Delete: {
-					if (!arg) throw language.get(LanguageKeys.Commands.Social.RemindmeDeleteNoId);
+					if (!arg) throw t(LanguageKeys.Commands.Social.RemindmeDeleteNoId);
```
```suggestion
					if (!arg) throw t(LanguageKeys.Commands.Social.RemindMeDeleteNoID);
```

vladfrangu commented on src/commands/Social/remindme.ts:
```diff
@@ -71,23 +70,23 @@ interface ReminderScheduledTask extends ScheduleEntity {
 						return task;
 					}
-					throw language.get(LanguageKeys.Commands.Social.RemindmeNotfound);
+					throw t(LanguageKeys.Commands.Social.RemindmeNotfound);
```
Ditto for this and everything else

vladfrangu commented on src/commands/Social/vault.ts:
```diff
@@ -100,8 +101,8 @@ export default class Vault extends SkyraCommand {
+	private async buildEmbed(message: KlasaMessage, t: TFunction, money: number, vault: number, coins?: number, hasDeposited = false) {
+		const { accountMoney, accountVault, depositedDescription, withdrewDescription, showDescription } = t(
 			LanguageKeys.Commands.Social.VaultEmbedData,
 			{ coins }
```
returnObjects?

vladfrangu commented on src/commands/Starboard/star.ts:
```diff
@@ -127,45 +124,30 @@ export default class extends SkyraCommand {
-								count: stars
-							}
-						)
+						t(LanguageKeys.Commands.Starboard.StarTopstarredDescription, {
```
Capitalizations

vladfrangu commented on src/commands/Starboard/star.ts:
```diff
@@ -127,45 +124,30 @@ export default class extends SkyraCommand {
-								count: stars
-							}
-						)
+						t(LanguageKeys.Commands.Starboard.StarTopreceiversDescription, {
```
Ditto

vladfrangu commented on src/commands/Suggestions/resolvesuggestion.ts:
```diff
@@ -34,26 +34,26 @@ const enum SuggestionsColors {
 			const { suggestions } = await DbSet.connect();
 			const suggestionData = await suggestions.findOne({ id, guildID: message.guild!.id });
-			if (!suggestionData) throw language.get(LanguageKeys.Commands.Suggestions.ResolveSuggestionIdNotFound);
+			if (!suggestionData) throw t(LanguageKeys.Commands.Suggestions.ResolveSuggestionIdNotFound);
```
```suggestion
			if (!suggestionData) throw t(LanguageKeys.Commands.Suggestions.ResolveSuggestionIDNotFound);
```

vladfrangu commented on src/commands/Suggestions/resolvesuggestion.ts:
```diff
@@ -115,7 +115,7 @@ export default class extends SkyraCommand {
 				await suggestionData.author!.send(messageContent, { embed: newEmbed });
 			} catch {
-				await message.channel.send(language.get(LanguageKeys.Commands.Suggestions.ResolveSuggestionDmFail));
+				await message.channel.send(t(LanguageKeys.Commands.Suggestions.ResolveSuggestionDmFail)!);
```
whats that `!` for

vladfrangu commented on src/commands/Suggestions/suggest.ts:
```diff
@@ -35,21 +35,21 @@ export default class extends SkyraCommand {
 			suggestionsChannel = this.client.channels.cache.get(suggestionsChannelID ?? '') as TextChannel | undefined;
 			if (!suggestionsChannel?.postable) {
-				throw language.get(LanguageKeys.Commands.Suggestions.SuggestNopermissions, {
+				throw t(LanguageKeys.Commands.Suggestions.SuggestNopermissions, {
```
Capitalization

vladfrangu commented on src/commands/System/Admin/reload.ts:
```diff
@@ -0,0 +1,66 @@
+	usage: '<Store:store|Piece:piece|everything:default>'
+})
+export default class extends SkyraCommand {
+	public async run(message: KlasaMessage, [piece]: [Piece | 'everything']) {
```
inaccurate type for Piece, it can be store too

vladfrangu commented on src/commands/System/donate.ts:
```diff
@@ -5,20 +5,20 @@ import { CommandStore, KlasaMessage } from 'klasa';
+		const t = await message.fetchT();
 		try {
-			const extended = (language.get(LanguageKeys.Commands.System.DonateExtended).extendedHelp as string[]).join('\n');
+			const extended = t(this.extendedHelp).extendedHelp!;
```
returnObjects?

vladfrangu commented on src/commands/Tools/Dictionary/define.ts:
```diff
@@ -57,15 +57,15 @@ export default class extends RichDisplayCommand {
 			);
 		} catch {
-			throw language.get(LanguageKeys.Commands.Tools.DefineNotfound);
+			throw t(LanguageKeys.Commands.Tools.DefineNotfound);
```
Capitalization

vladfrangu commented on src/commands/Tools/Websearch/dsearch.ts:
```diff
@@ -21,13 +21,13 @@ export default class extends SkyraCommand {
 
 		if (body.RelatedTopics && body.RelatedTopics.length > 0) {
-			embed.addField(await message.fetchLocale(LanguageKeys.Commands.Tools.DuckDuckGoLookalso), body.RelatedTopics[0].Text);
+			embed.addField(await message.resolveKey(LanguageKeys.Commands.Tools.DuckDuckGoLookalso), body.RelatedTopics[0].Text);
```
Capitalization

vladfrangu commented on src/commands/Tools/Websearch/itunes.ts:
```diff
@@ -5,53 +5,50 @@ import { GuildMessage } from '#lib/types';
 	cooldown: 10,
-	description: (language) => language.get(LanguageKeys.Commands.Tools.ItunesDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Tools.ItunesExtended),
+	description: LanguageKeys.Commands.Tools.ItunesDescription,
```
n/a but this hurts

vladfrangu commented on src/commands/Tools/Websearch/movies.ts:
```diff
@@ -48,27 +45,27 @@ export default class extends RichDisplayCommand {
-		const fieldsData = language.get(LanguageKeys.Commands.Tools.MoviesData);
+	private async buildDisplay(message: GuildMessage, t: TFunction, movies: Tmdb.TmdbMovieList['results']) {
+		const titles = t(LanguageKeys.Commands.Tools.MoviesTitles);
+		const fieldsData = t(LanguageKeys.Commands.Tools.MoviesData);
```
returnObject?

vladfrangu commented on src/commands/Tools/Websearch/shows.ts:
```diff
@@ -48,27 +45,27 @@ export default class extends RichDisplayCommand {
-		const fieldsData = language.get(LanguageKeys.Commands.Tools.ShowsData);
+	private async buildDisplay(message: GuildMessage, t: TFunction, shows: Tmdb.TmdbSeriesList['results']) {
+		const titles = t(LanguageKeys.Commands.Tools.ShowsTitles);
+		const fieldsData = t(LanguageKeys.Commands.Tools.ShowsData);
```
returnObjects?

vladfrangu commented on src/commands/Tools/Websearch/wikipedia.ts:
```diff
@@ -4,31 +4,32 @@ import { LanguageKeys } from '#lib/types/namespaces/LanguageKeys';
 
 		if (text.query.pageids[0] === '-1') {
-			throw language.get(LanguageKeys.Commands.Tools.WikipediaNotfound);
+			throw t(LanguageKeys.Commands.Tools.WikipediaNotfound);
```
capitalization

vladfrangu commented on src/commands/Tools/Websearch/youtube.ts:
```diff
@@ -21,8 +21,8 @@ export default class extends SkyraCommand {
-			description: (language) => language.get(LanguageKeys.Commands.Tools.YoutubeDescription),
-			extendedHelp: (language) => language.get(LanguageKeys.Commands.Tools.YoutubeExtended),
+			description: LanguageKeys.Commands.Tools.YoutubeDescription,
+			extendedHelp: LanguageKeys.Commands.Tools.YoutubeExtended,
```
YouTube :smug:

vladfrangu replied:
I wasn't memeing

vladfrangu commented on src/commands/Tools/Websearch/youtube.ts:
```diff
@@ -37,7 +37,7 @@ export default class extends SkyraCommand {
 		const results = data.items.slice(0, 5);
 
-		if (!results.length) throw await message.fetchLocale(LanguageKeys.Commands.Tools.YoutubeNotfound);
+		if (!results.length) throw await message.resolveKey(LanguageKeys.Commands.Tools.YoutubeNotfound);
```
ditto

vladfrangu commented on src/commands/Tools/topinvites.ts:
```diff
@@ -32,21 +29,21 @@ export default class extends RichDisplayCommand {
 				.setColor(await DbSet.fetchColor(message))
 		);
-		const embedData = language.get(LanguageKeys.Commands.Tools.TopInvitesEmbedData);
+		const embedData = t(LanguageKeys.Commands.Tools.TopInvitesEmbedData);
```
returnObject?

vladfrangu commented on src/commands/Tools/whois.ts:
```diff
@@ -58,9 +58,13 @@ export default class extends SkyraCommand {
-		const fields = language.get(LanguageKeys.Commands.Tools.WhoisMemberFields, { member });
+	private member(t: TFunction, member: GuildMember) {
+		const titles = t(LanguageKeys.Commands.Tools.WhoisMemberTitles);
+		const fields = t(LanguageKeys.Commands.Tools.WhoisMemberFields, {
```
returnObjects?

vladfrangu commented on src/commands/Twitch/twitch.ts:
```diff
@@ -3,25 +3,26 @@ import { CdnUrls } from '#lib/types/Constants';
 
-		const titles = language.get(LanguageKeys.Commands.Twitch.TwitchTitles);
-		const affiliateStatus = this.parseAffiliateProgram(language, channel.broadcaster_type);
+		const titles = t(LanguageKeys.Commands.Twitch.TwitchTitles);
```
returnObjects?

vladfrangu commented on src/commands/Twitch/twitchsubscription.ts:
```diff
@@ -242,55 +234,41 @@ export default class extends SkyraCommand {
-		const [guildSubscriptions, language] = await message.guild.readSettings((settings) => [settings[this.#kSettingsKey], settings.getLanguage()]);
+		const t = await message.fetchT();
+
+		const guildSubscriptions = await message.guild.readSettings((settings) => settings[this.#kSettingsKey]);
```
unite

vladfrangu commented on src/commands/Weeb/wbanghead.ts:
```diff
@@ -3,8 +3,8 @@ import { LanguageKeys } from '#lib/types/namespaces/LanguageKeys';
-	description: (language) => language.get(LanguageKeys.Commands.Weeb.BangheadDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Weeb.BangheadExtended),
+	description: LanguageKeys.Commands.Weeb.BangheadDescription,
+	extendedHelp: LanguageKeys.Commands.Weeb.BangheadExtended,
```
BangHead

vladfrangu commented on src/commands/Weeb/wthumbsup.ts:
```diff
@@ -3,8 +3,8 @@ import { LanguageKeys } from '#lib/types/namespaces/LanguageKeys';
-	description: (language) => language.get(LanguageKeys.Commands.Weeb.ThumbsupDescription),
-	extendedHelp: (language) => language.get(LanguageKeys.Commands.Weeb.ThumbsupExtended),
+	description: LanguageKeys.Commands.Weeb.ThumbsupDescription,
+	extendedHelp: LanguageKeys.Commands.Weeb.ThumbsupExtended,
```
ThumbsUp

vladfrangu commented on src/events/mentionSpamExceeded.ts:
```diff
@@ -17,14 +17,14 @@ export default class extends Event {
 			nms.delete(message.author.id);
 
-			const reason = language.get(LanguageKeys.Monitors.NmsModlog, { threshold });
+			const reason = t(LanguageKeys.Monitors.NmsModlog, { threshold });
```
NmsModLog

vladfrangu commented on src/events/music/musicSongVolumeUpdateNotify.ts:
```diff
@@ -9,16 +9,17 @@ export default class extends AudioEvent {
+		return t(LanguageKeys.Commands.Music.VolumeChangedExtreme, {
 			emoji: '📢',
-			text: pickRandom(language.get(LanguageKeys.Commands.Music.VolumeChangedTexts)),
+			text: pickRandom(t(LanguageKeys.Commands.Music.VolumeChangedTexts)),
```
returnObjects?

vladfrangu commented on src/lib/database/entities/GiveawayEntity.ts:
```diff
@@ -207,34 +210,29 @@ export class GiveawayEntity extends BaseEntity {
+					: t(LanguageKeys.Giveaway.EndedNoWinner);
 			case States.LastChance:
-				return language.get(LanguageKeys.Giveaway.Lastchance, { time: this.remaining });
+				return t(LanguageKeys.Giveaway.Lastchance, { time: this.remaining });
```
LastChance

vladfrangu commented on src/lib/database/entities/GiveawayEntity.ts:
```diff
@@ -277,14 +275,14 @@ export class GiveawayEntity extends BaseEntity {
+				return t(LanguageKeys.Giveaway.EndedTitle);
 			case States.LastChance:
-				return language.get(LanguageKeys.Giveaway.LastchanceTitle);
+				return t(LanguageKeys.Giveaway.LastchanceTitle);
```
LastChance

vladfrangu commented on src/lib/extensions/SkyraDMChannel.ts:
```diff
@@ -1,14 +1,11 @@
-		const language = this.client.languages.get(languageKey);
-		if (language) return language;
-		throw new Error(`The language '${language}' is not available.`);
+		const lang: string = await this.client.fetchLanguage(({ channel: this, guild: null } as unknown) as Message);
```
cast as any works too, shorter

vladfrangu commented on src/lib/extensions/SkyraGuild.ts:
```diff
@@ -19,15 +20,17 @@ export class SkyraGuild extends Structures.get('Guild') {
-	public fetchLanguage(this: Guild): Promise<Language> {
-		return this.readSettings((entity) => this.client.languages.get(entity.language)!);
+	public async fetchLanguage() {
+		const lang: string = await this.client.fetchLanguage(({ guild: this, channel: null } as unknown) as Message);
```
bonk

vladfrangu commented on src/lib/extensions/SkyraNewsChannel.ts:
```diff
@@ -1,13 +1,10 @@
-		const language = this.client.languages.get(languageKey);
-		if (language) return language;
-		throw new Error(`The language '${language}' is not available.`);
+		const lang: string = await this.client.fetchLanguage(({ channel: this, guild: this.guild } as unknown) as Message);
```
same

vladfrangu commented on src/lib/extensions/SkyraTextChannel.ts:
```diff
@@ -1,10 +1,14 @@
 
 export class SkyraTextChannel extends TextBasedExtension(Structures.get('TextChannel')) {
+	public async fetchLanguage() {
+		const lang: string = await this.client.fetchLanguage(({ channel: this, guild: this.guild } as unknown) as Message);
```
bork

vladfrangu commented on src/lib/games/base/BaseReactionGame.ts:
```diff
@@ -27,10 +27,10 @@ export abstract class BaseReactionGame<T> extends BaseGame<T> {
 	protected async onStart(): Promise<unknown> {
 		try {
-			this.message = await this.message.send(pickRandom(this.language.get(LanguageKeys.System.Loading)));
+			this.message = await this.message.send(pickRandom(this.t(LanguageKeys.System.Loading)));
```
returnObjects?

vladfrangu commented on src/lib/structures/SelfModerationCommand.ts:
```diff
@@ -105,13 +107,13 @@ export abstract class SelfModerationCommand extends Command {
 				const softAction = kSoftActions.get(arg.toLowerCase());
 				if (typeof softAction === 'undefined') {
-					throw await message.fetchLocale(LanguageKeys.Commands.Moderation.AutomaticParameterInvalidSoftaction, { name: this.name });
+					throw await message.resolveKey(LanguageKeys.Commands.Moderation.AutomaticParameterInvalidSoftaction, { name: this.name });
```
SoftAction >:C

vladfrangu commented on src/lib/structures/SelfModerationCommand.ts:
```diff
@@ -121,7 +123,7 @@ export abstract class SelfModerationCommand extends Command {
 				const hardAction = kHardActions.get(arg.toLowerCase());
 				if (typeof hardAction === 'undefined') {
-					throw await message.fetchLocale(LanguageKeys.Commands.Moderation.AutomaticParameterInvalidHardaction, { name: this.name });
+					throw await message.resolveKey(LanguageKeys.Commands.Moderation.AutomaticParameterInvalidHardaction, { name: this.name });
```
HardAction~ >:3

vladfrangu commented on src/lib/structures/SettingsMenu.ts:
```diff
@@ -49,7 +49,7 @@ export class SettingsMenu {
 	public async init(): Promise<void> {
 		this.response = (await this.message.send(
-			new MessageEmbed().setColor(BrandingColors.Secondary).setDescription(pickRandom(this.language.get(LanguageKeys.System.Loading)))
+			new MessageEmbed().setColor(BrandingColors.Secondary).setDescription(pickRandom(this.t(LanguageKeys.System.Loading)))
```
returnObjects?

vladfrangu commented on src/lib/util/Games/Slotmachine.ts:
```diff
@@ -129,7 +129,7 @@ export class Slotmachine {
 			.setTextAlign('right')
 			.printText(
-				lang.get(playerHasWon ? LanguageKeys.Commands.Games.SlotmachineCanvasTextWon : LanguageKeys.Commands.Games.SlotmachineCanvasTextLost),
+				lang(playerHasWon ? LanguageKeys.Commands.Games.SlotmachineCanvasTextWon : LanguageKeys.Commands.Games.SlotmachineCanvasTextLost),
```
SlotMachine

vladfrangu commented on src/lib/util/Language/list.ts:
```diff
@@ -0,0 +1,20 @@
+ * @param values The values to list
+ * @param conjunction The conjunction to use between the values
+ */
+export function list(values: readonly string[], conjunction: string) {
```
Is this even needed anymore?

vladfrangu commented on src/lib/util/Security/ModerationActions.ts:
```diff
@@ -425,23 +425,23 @@ export class ModerationActions {
-					? language.get(LanguageKeys.Commands.Moderation.ActionUnSoftbanReason, { reason: moderationLog.reason! })
-					: language.get(LanguageKeys.Commands.Moderation.ActionUnSoftbanNoReason)
+					? t(LanguageKeys.Commands.Moderation.ActionUnSoftbanReason, { reason: moderationLog.reason! })
+					: t(LanguageKeys.Commands.Moderation.ActionUnSoftbanNoReason)
```
SoftBan

vladfrangu commented on src/lib/util/Security/ModerationActions.ts:
```diff
@@ -937,15 +931,15 @@ export class ModerationActions {
-		const language = await this.guild.fetchLanguage();
-		const actions = language.get(LanguageKeys.Commands.Moderation.moderationActions);
+		const t = await this.guild.fetchT();
+		const actions = t(LanguageKeys.Commands.Moderation.moderationActions);
```
Any reason moderation isn't capitalized?

vladfrangu reviewed (changes requested):
Ignore any returnObjects bc apparently y'all default it to true sooo

kyranet committed:
chore: requested changes

kyranet committed:
fix: fixes on fixes within fixes

kyranet committed:
refactor: use more Intl

NM-EEA-Y committed:
New Crowdin updates (#1456)

kyranet committed:
refactor: make level and profile Intl aware

NM-EEA-Y committed:
New Crowdin updates (#1458)

Co-authored-by: Antonio Román <kyradiscord@gmail.com>

kyranet committed:
refactor: more Intl

favna committed:
feat: add constants for nl-NL

favna committed:
chore: we know the game and we're gonna play it

NM-EEA-Y committed:
New Crowdin updates (#1459)

Co-authored-by: Jeroen Claassens <support@favware.tech>

Quantumlyy committed:
feat: add more precise i18next options (#1461)

kyranet committed:
fix: more and more fixes

kyranet committed:
fix: remove language limit

kyranet committed:
fix: add missing line in SetMigrations.sql

NM-EEA-Y committed:
New Crowdin updates (#1460)

kyranet committed:
fix: resolved mismatching ns bug

kyranet committed:
fix: added missing namespaces

kyranet committed:
fix: casing

vladfrangu commented on src/commands/GameIntegration/overwatch.ts:
```diff
@@ -152,7 +151,7 @@ export default class extends RichDisplayCommand {
 		} = overwatchData[type];
 
 		const timePlayedMilliseconds = Number(timePlayed.split(':')[0]) * Time.Hour + Number(timePlayed.split(':')[1]) * Time.Minute;
```
n/a but the double split can be optimized

vladfrangu commented on src/commands/Pokemon/pokedex.ts:
```diff
@@ -151,17 +151,17 @@ export default class extends RichDisplayCommand {
 		const evoChain = this.getEvoChain(pokeDetails);
-		const embedTranslations = language.get(LanguageKeys.Commands.Pokemon.PokedexEmbedData, {
+		const embedTranslations = t(LanguageKeys.Commands.Pokemon.PokedexEmbedData, {
 			otherFormes: pokeDetails.otherFormes ?? [],
```
n/a but fav pls. It's Forms :megu:

vladfrangu commented on src/commands/Social/Profile Management/banner.ts:
```diff
@@ -34,23 +34,23 @@ export default class extends SkyraCommand {
+		if (!accepted) throw t(LanguageKeys.Commands.Social.BannerPaymentCancelled);
 
-		if (author.money < banner.price) throw language.get(LanguageKeys.Commands.Social.BannerMoney, { money: author.money, cost: banner.price });
+		if (author.money < banner.price) throw t(LanguageKeys.Commands.Social.BannerMoney, { money: author.money, cost: banner.price });
```
Any reason you check twice?

kyranet replied:
Same thing as pay, you buy a banner, get prompted, buy another, get prompted, accept both, and you'll get two banners and negative currency.

This way people can't duplicate money.

At least, this was with SettingsGateway, it doesn't hold true anymore, lol.

vladfrangu commented on src/commands/Social/autorole.ts:
```diff
@@ -71,58 +71,58 @@ export default class extends SkyraCommand {
 
 			if (roleIndex === -1) {
-				throw language.get(LanguageKeys.Commands.Social.AutoRoleUpdateConfigured);
+				throw t(LanguageKeys.Commands.Social.AutoRoleUpdateConfigured);
```
Technically this isn't the right message for a missing entry. Use the same one as in update

vladfrangu commented on src/commands/Tags/tag.ts:
```diff
@@ -41,87 +41,86 @@ export default class extends SkyraCommand {
 	public async list(message: GuildMessage) {
 		// Get tags, prefix, and language
-		const [[tags, prefix], language] = await Promise.all([
+		const [[tags, prefix], t] = await Promise.all([
```
Any reason this doesn't use guild.readSettings(settinga => [])?

kyranet replied:
Who was the heretic person who did this?

vladfrangu commented on src/commands/Twitch/twitchsubscription.ts:
```diff
@@ -193,31 +189,27 @@ export default class extends SkyraCommand {
+			status === NotificationsStreamsTwitchEventStatus.Offline
+				? LanguageKeys.Commands.Twitch.TwitchSubscriptionRemoveSuccessOffline
+				: LanguageKeys.Commands.Twitch.TwitchSubscriptionRemoveSuccessLive,
+			[{ name: streamer.display_name, channel: channel.name }]
```
Any reason for channel.name here and channel.tostring() everywhere else

vladfrangu reviewed (dismissed):
Ohh what i do to make kyra happy... Continuing at home

![Screenshot_2021-01-12-15-24-33-612_com chrome canary](https://user-images.githubusercontent.com/17960496/104328609-a8762300-54f4-11eb-9da7-4fdf317ca10f.jpg)

kyranet committed:
fix: resolved even more bug requests

favna committed:
chore: remove unused tokens from config

favna committed:
chore: fix some bugs

favna committed:
fix: properly capitilize YouTube

favna committed:
Merge branch 'main' into feat/i18next

vladfrangu commented on src/events/mentionSpamExceeded.ts:
```diff
@@ -17,14 +17,14 @@ export default class extends Event {
-			await message
-				.sendLocale(LanguageKeys.Monitors.NmsMessage, [{ user: message.author }])
+			await message.channel
+				.sendTranslated(LanguageKeys.Monitors.NoMentionSpamMessage, [{ user: message.author }])
```
Re-use t, also is user meant to be the mention or?
```suggestion
			await message.channel
				.send(t(LanguageKeys.Monitors.NoMentionSpamMessage, { user: message.author.toString() }))
```

vladfrangu commented on src/lib/util/Games/Slotmachine.ts:
```diff
@@ -102,16 +102,16 @@ export class Slotmachine {
 
-		const lang = await this.message.fetchLanguage();
-		if (amount < 0) throw lang.get(LanguageKeys.Commands.Games.GamesCannotHaveNegativeMoney);
+		const lang = await this.message.fetchT();
```
Inconsistent naming (yes this is a priority of a nanometer give me a break)

favna committed:
chore: set proper klasa version and bump plugin

favna committed:
chore: requested changes

vladfrangu reviewed (dismissed):
JUST MERGE THIS MY SINS HAVE BEEN RENEWED

favna committed:
chore: grmbl
